custom-datetimes/custom_datetimes.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+"""custom_datetimes.ipynb
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/17T--fqnI3yZClgRX-AAWJUuJEUDIgvCx
+"""
+
+#regular expression way
+from datetime import datetime
+import pytz
+import re #regular expression to capture certain pattern in a data
+from datetime import timezone
+
+#part 1
+ex1 = "03/23/21"
+p1 = datetime.strptime(ex1, "%m/%d/%y")
+p1
+
+#part 2
+ex2 = "23/03/2021"
+p2 = datetime.strptime(ex2, "%d/%m/%Y")
+p2
+
+#part 3
+ex3 = "March 23rd, 2021 13:01 US/Pacific"
+ex3_list = ex3.split()
+tz = ex3_list[-1]
+tz = pytz.timezone(tz)
+ex3_new = " ".join(ex3_list[:-1])
+
+p3 = datetime.strptime(ex3_new, "%B %drd, %Y %H:%M").astimezone(tz)
+p3
+
+#part 4
+ex4 = "1:01pm 23rd March, 2021 Europe/London"
+ex4_list = ex4.split()
+tz = ex4_list[-1]
+tz = pytz.timezone(tz)
+ex4_new = " ".join(ex4_list[:-1])
+
+p4 = datetime.strptime(ex4_new, "%I:%M%p %drd %B, %Y").astimezone(tz)
+p4
+
+#part 5
+ex5 = 1616482800
+p5 = datetime.fromtimestamp(ex5).strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
You still want to be parsing this as a python datetime object (right now you are returning a string)